### PR TITLE
DHCP DUID parsing fix, logging fix, man-page fix

### PR DESCRIPTION
**Pull request by vinaykul**
This change addresses the latest review items by @poettering , @keszybz for https://github.com/systemd/systemd/pull/2890 
1. Replace strtol with unhexchar, verified with valid and invalid DUID strings.
2. Fix logging to use log_syntax instead of log_error.
3. Fix man-pages to use markup, remove options that are not yet implemented.
4. Remove spurious header line in files.

### CODE REVIEW

**Comment by keszybz** [comment]
This text in the PR should really be in the commit message.

**Comment by keszybz** [comment]
I think that the man page now doesn't match the parsing code which accepts "raw", "link-layer", etc.
And we have a `Type=` setting with only one allowed option, and with a default which matches this one allowed option, i.e. the same situation I complained about a while ago :)

I think the best solution would be to add a second patch on top (separate), which drops `Type=` description from the man page. When/if we get to adding support for other `Type=` values, we'll revert that patch and continue from there.

Otherwise, patch looks good.

**Comment by vinaykul** [comment]
@keszybz I agree that hiding is Type field from the docs is a good stop-gap measure at this time since it is truly optional right now with the default to 'raw' and mentioning it is unnecessary. 

The other values that Type can take without complaining are also not documented so a user will have to go the extra length of reading the code to assign type values only to find it unusable. 

I'll prepare the man-page diff and stack it on top of this on a short while. Hopefully I'll get around to completing the story on this once I'm done with the DUID/IAID query change.

File: src/network/networkd-conf.c [hunk 1]
```diff
@@ -152,4 +161,11 @@ int config_parse_duid_rawdata(
         *dhcp_duid_len = duid_index;
 
         return 0;
+
+fail:
+        /* Ignore invalid DUID assignment */
+        *duid_type = _DUID_TYPE_INVALID;
+        *dhcp_duid_len = 0;
+        *dhcp_duid_type = 0;
+        return 0;
```
Comment (keszybz) [inline src/network/networkd-conf.c:163]: I don't like this chunk. If we say "ignoring", we shouldn't set a value. In particular, if the config file has:

```
RawData=ab:bc:de
RawData=something-invalid
```

the value would be reset to `_DUID_TYPE_INVALID` unexpectedly.

Comment (vinaykul) [inline src/network/networkd-conf.c:163]: Indeed.. Thanks again for the sharp review. I'll work on a fix to hold on to the 'last-known-good' value.

Comment (keszybz) [inline src/network/networkd-conf.c:163]: The general rule in systemd (99.9% of cases) is "don't modify anything on failure, let the caller deal with that". Having things behave consistently in this way really helps avoid surprising side effects.

**Comment by vinaykul** [comment]
For the DUID configuration interface, the two commits above addresses the outstanding review items by @poettering @keszybz except for having [DUID] section vs. [DHCP] section.

@teg Please take a look at Lennart's comment https://github.com/systemd/systemd/pull/2890#discussion_r58233311 and let me know if we want to keep [DUID] or if we should instead go with [DHCP] DUIDType, DUIDRawData ... 

This is a fairly easy change to make, but just want to get us all on the same page before I work on it.

**Comment by poettering** [comment]
So, I had another closer look at this, and I still don't like the current state of things. I posted a new issue in #3127 listening the things that still need fixing. I also pepped PR #3126 that fixes some other issues I see.

**Pull request merged**